src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -601,6 +596,9 @@ private void finish(@Nonnull Result r, @CheckForNull Throwable t) {
                 }
                 listener = null;
             }
+            synchronized (getMetadataGuard()) {
+                completed = true;
```

Comment:
Moving this around doesn't actually matter as discussed in https://github.com/jenkinsci/workflow-job-plugin/pull/131/files#r291657569, right? If not, I'd just move it back to where it was originally.

Reply:
Mm, can try it. Means going back through a bunch of rounds of integration testing, unfortunately.